lib/cylc/task_pool.py

```diff
@@ -1224,6 +1224,9 @@ def process_queued_db_ops(self):
         """Handle queued db operations for each task proxy."""
         for itask in self.get_all_tasks():
             # (runahead pool tasks too, to get new state recorders).
+            if not (any(itask.db_inserts_map.values()) or
+                    any(itask.db_updates_map.values())):
+                continue
```

Comment:
Won't this still result in some unnecessary work if one but not both of the maps is empty?  You could do:
```
if any(itask.db_inserts_map.values()):
    for table_name, db_inserts in sorted(itask.db_inserts_map.items()):
         #...
if any(itask.db_updates_map.values()):
    for table_name, db_updates in sorted(itask.db_updates_map.items()):
         #...
```